common/include/common/time.h

```diff
@@ -61,8 +61,8 @@ static inline void nsleep(uint64_t ns)
 #else
   const struct timespec ts =
   {
-    .tv_sec  = ns / 1e9,
-    .tv_nsec = ns - ((ns / 1e9) * 1e9)
+    .tv_sec  = ns / (time_t)1000000000L,
```

Comment:
Why the `(time_t)` cast? If you use the literal without the `L` prefix, it'll be promoted to the same size as the other operand.
```suggestion
    .tv_sec  = ns / 1000000000,
```